meson.build

```diff
@@ -183,10 +183,13 @@ usr_deps = [kmod_dep, crypto_dep, mount_dep, sysdeps_dep, fwparam_dep, libiscsi_
 if not no_systemd
   usr_deps += systemd_dep
 endif
+if get_option('isns').enabled()
+  genl_cargs += '-DISNS_SUPPORTED'
+endif
```

Comment:
You should instead do this when `cc.find_library('isns', required: get_option('isns')).found()` is true. Currently, it defaults to enabled, but users can set it to either disabled or "auto". If they set it to auto, then this define will cause the headers to be used and code which calls into it to be *used*, it just won't actually link to the needed library.

Reply:
So I tested with and without iSNS installed on my system, using "auto". It works perfectly. As I said, if you pass in "auto", "enabled()" becomes "true" or "false" based on whether or not it finds the library. So, in the case of "auto", if the iSNS library isn't found, "get_option('isns').enabled()" becomes false.

Reply:
Sorry, I got this backwards -- but the code is still wrong.

I see this sort of issue in various places, and open-iscsi isn't even the only place this week that I've seen it. I may have gotten mixed up with the logic flow of another project.

But the root of the problem is this: using methods of a *feature* option cannot tell you whether a library is found. You're using code that asks "was this passed as `enabled`?" to toggle the define. The return value for enabled() is always the same regardless of any found libraries. What happens is this. If the user passes "auto":
- it is not enabled, so this code evaluates to boolean "false" and the define is ***not*** passed.
- user actually does have isns installed
- cc.find_library succeeds, isns is passed in usr_deps
- the final application links to isns, but doesn't use it

The result of the mistake isn't a failure to build properly. Instead, it's just not possible to auto detect support.

Reply:
Since I've changed the meson script, as suggested, I'm going to mark this conversation as resolved. Thanks again for all of your feedback!

Reply:
Sure, no problem. :)